File: April 2023/26th April - Seating Arrangement/26thApril.py
class Solution:
    def is_possible_to_get_seats(self, n : int, m : int, seats : List[int]) -> bool:
        for i in range(m):
            if n == 0:
                return True
            if seats[i] == 0 and (i == 0 or seats[i-1] == 0) and (i == m-1 or seats[i+1] == 0):
                seats[i] = 1
                n -= 1
                
        return n == 0

File: April 2023/26th April - Seating Arrangement/26thApril.cpp
class Solution{
    public:
    bool is_possible_to_get_seats(int n, int m, vector<int>& seats){
        // Write your code here.
        for (int i=0; i<m; i++) {
            if (n == 0) return true;
            if (seats[i] == 0 && (i == 0 || seats[i-1] == 0) && (i == m-1 || seats[i+1] == 0)) {
                seats[i] = 1;
                n -= 1;
            }
        }
        return n == 0;
    }
};